### ISSUE TITLE
Trying to get remote debugging working

### ISSUE THREAD

**Issue by MatsIBengtsson** [inkstitch/inkstitch#521]
Found the debug module, which described remote debugging. As I understood it, include "DEBUG" in root, and a debug configuration is created. Looking in the log, it might have happened, there is a green line about debug. Seemingly giving me the port number for the debug as well, possibly requiring SSH (isch, hope not).

Then "start any extension". So I start incscape and run an extension. It runs without stopping. Is there more I need to do to get the debug activated in Ink/Stitch itself?

My debugger is waiting and not seeing any connection. If there is more to do on the Ink/Stitch side, then that will not happen. So is there more I need to do?

**Comment by lexelby** [comment]
It's designed to work with PyDev, and I don't think it's compatible with IDEA.  I really recommend pydev, at least for debugging.  It's saved me a ton of time.

When the DEBUG file exists,Ink/Stitch will try to connect to PyDev, and if it can't, it will continue on without debugging.  You do still get debug logging, which can be useful.  You can even generate SVG files for debugging which was really useful when developing the auto-fill algorithm.

**Comment by MatsIBengtsson** [comment]
> It's designed to work with PyDev, and I don't think it's compatible with IDEA. I really recommend pydev, at least for debugging. It's saved me a ton of time.
> 
> When the DEBUG file exists,Ink/Stitch will try to connect to PyDev, and if it can't, it will continue on without debugging. You do still get debug logging, which can be useful. You can even generate SVG files for debugging which was really useful when developing the auto-fill algorithm.

PyCharm (which I am using) is using pydevd as well. default is localhost port 0, I changed it to localhost:9999. But I am on windows, and SSH is a struggle. I can download and try PyDev, not a problem.

There is no SSH involved? I saw both the 9999:localhost:9999 and ssh key, which got me thinking it might be.

**Comment by MatsIBengtsson** [comment]
> PyCharm (which I am using) is using pydevd as well. default is localhost port 0, I changed it to localhost:9999. But I am on windows, and SSH is a struggle. I can download and try PyDev, not a problem.
> 
> There is no SSH involved? I saw both the 9999:localhost:9999 and ssh key, which got me thinking it might be.

If I add "DEBUG" subdirectory at incscape/extension (where incstitich.py is installed), I do get a debug.log at incscape/extension/inkstitch/bin. It says pydevd import failed. Even when I place pydevd.py file in both same directory as inkstitch.py and in the bin directory (where inkstitch.exe is).

Should not pydevd come as part of the pyinstaller?

**Comment by lexelby** [comment]
SSH definitely isn't needed.  This isn't real "remote" debugging, the "remote" part just means that the debugger is a separate process (your IDE).  PyDev listens on port 5678, and I bet that's probably the default.

PyInstaller isn't packaging in pydevd because we're not installing it with our `requirements.txt`.  I have never tried debugging a pyinstaller-packaged version of Ink/Stitch, and I'm not even sure it would work.  I always debug on my local development clone so that the debugger can find Ink/Stitch's source code.

**Comment by MatsIBengtsson** [comment]
> SSH definitely isn't needed. This isn't real "remote" debugging, the "remote" part just means that the debugger is a separate process (your IDE). PyDev listens on port 5678, and I bet that's probably the default.

Thank you, yes, I got it working locally with 5678, and it is the default. But then pyinstaller claimed I used relative naming, and it should better be absolute (to find the sources). So I wanted to get it working using your real build. And from the command, it looked like it used 9999.

> PyInstaller isn't packaging in pydevd because we're not installing it with our `requirements.txt`. I have never tried debugging a pyinstaller-packaged version of Ink/Stitch, and I'm not even sure it would work. I always debug on my local development clone so that the debugger can find Ink/Stitch's source code.

It does work. There are some issues, like inkstitch.py not being same as the one we package from, thus showing incorrect lines. And finding the right source to show. I did download the source code from the server after build, so I think it would work that way as well. Anyway, if you never tried it that way, no reason I should do it.

I need to use the loop input (which creates an svg) to Output (which writes from the elements). Looked to me easiest way would be to debug the exe, ran by incscape to achieve that loop?
